pkg/container-collection/k8s.go

```diff
@@ -234,10 +234,19 @@ func getContainerRuntimeSocketPath(clientset *kubernetes.Clientset, nodeName str
 	if err != nil {
 		return "", fmt.Errorf("getting /configz: %w", err)
 	}
-	socketPath, found := strings.CutPrefix(kubeletConfig.ContainerRuntimeEndpoint, "unix://")
-	if !found {
-		return "", fmt.Errorf("socket path does not start with unix://")
+
+	endpoint := kubeletConfig.ContainerRuntimeEndpoint
+	socketPath := endpoint
+
+	// If it starts with unix://, strip the prefix
+	if strings.HasPrefix(endpoint, "unix://") {
+		socketPath = strings.TrimPrefix(endpoint, "unix://")
+	}
+
+	if socketPath == "" {
+		return "", fmt.Errorf("container runtime socket path is empty")
 	}
```

Comment:
What about something like:

```suggestion
	socketPath := strings.TrimPrefix(kubeletConfig.ContainerRuntimeEndpoint, "unix://")
	socketPath == "" {
		return "", fmt.Errorf("container runtime socket path is empty")
	}
```